Tests/UnitTests/Networking/ETagManagerTests.swift

```diff
@@ -212,17 +286,214 @@ class ETagManagerTests: TestCase {
+        let url = self.urlForTest()
+        let request = URLRequest(url: url)
+        let cacheKey = url.absoluteString
+
+        let actualResponse = "response".asData
+
+        self.mockUserDefaults.mockValues[cacheKey] = ETagManager.Response(
+            eTag: eTag,
+            statusCode: .success,
+            data: actualResponse,
+            validationResult: .notRequested
+        ).asData()
+
+        let response = self.eTagManager.eTagHeader(for: request)
+        expect(response[ETagManager.eTagHeaderName]) == eTag
+    }
+
+    func testCachedResponseWithNoValidationResultIsIgnored() {
+        let eTag = "the_etag"
+        let url = self.urlForTest()
+        let request = URLRequest(url: url)
+        let cacheKey = url.absoluteString
+
+        let actualResponse = "response".asData
+
+        self.mockUserDefaults.mockValues[cacheKey] = """
+        {
+        "e_tag": "\(eTag)",
+        "status_code": 200,
+        "data": "\(actualResponse.asFetchToken)"
+        }
+        """.asData
+
+        let response = self.eTagManager.httpResultFromCacheOrBackend(
+            with: self.responseForTest(url: url,
+                                       body: nil,
+                                       eTag: eTag,
+                                       statusCode: .notModified),
+            request: request,
+            retried: false
+        )
+        expect(response).to(beNil())
+    }
+
+    func testCachedResponseIsFoundIfValidationWasNotRequested() {
+        let eTag = "the_etag"
+        let url = self.urlForTest()
+        let request = URLRequest(url: url)
+        let cacheKey = url.absoluteString
+
+        let actualResponse = "response".asData
+
+        self.mockUserDefaults.mockValues[cacheKey] = ETagManager.Response(
+            eTag: eTag,
+            statusCode: .success,
+            data: actualResponse,
+            validationResult: .notRequested
+        ).asData()
+
+        let response = self.eTagManager.httpResultFromCacheOrBackend(
+            with: self.responseForTest(url: url,
+                                       body: nil,
+                                       eTag: eTag,
+                                       statusCode: .notModified),
+            request: request,
+            retried: true
+        )
+        expect(response).toNot(beNil())
+        expect(response?.statusCode) == .success
+        expect(response?.body) == actualResponse
+        expect(response?.validationResult) == .notRequested
+    }
+
+    func testCachedResponseIsFoundIfValidationSucceeded() {
+        let eTag = "the_etag"
+        let url = self.urlForTest()
+        let request = URLRequest(url: url)
+        let cacheKey = url.absoluteString
+
+        let actualResponse = "response".asData
+
+        self.mockUserDefaults.mockValues[cacheKey] = """
+        {
+        "e_tag": "\(eTag)",
+        "status_code": 200,
+        "data": "\(actualResponse.asFetchToken)",
+        "validation_result": 1
```

Comment:
nit: maybe consider `"validation_result": \(.validated.rawValue)`

Reply:
In this particular case I wanted to hardcore it so we never break backwards compatibility if somebody changes the associated value. I'll add a comment to explain.

Reply:
Actually I'm adding plenty of test coverage for that serialization in #2277 so I'll follow your suggestion 👍🏻